central/reports/service/v2/service_impl.go

```diff
@@ -0,0 +1,71 @@
+package v2
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/reportconfigurations/service/common"
+	metadataDS "github.com/stackrox/rox/central/reports/metadata/datastore"
+	apiV2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/search"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type serviceImpl struct {
+	apiV2.UnimplementedReportServiceServer
+	metadataDatastore metadataDS.DataStore
+}
+
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		apiV2.RegisterReportServiceServer(grpcServer, s)
+	}
+}
+
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		return apiV2.RegisterReportServiceHandler(ctx, mux, conn)
+	}
+	return nil
+}
+
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, common.Authorizer.Authorized(ctx, fullMethodName)
```

Comment:
Implement authorization check for `/v2.ReportService/GetReportStatus` and `/v2.ReportService/GetReportLastStatus`.

Reply:
Also, add a unit test for the authorization check.